### PR TITLE
Api extensions

### LINKED ISSUE #1653: API Extensions
We should add support for our new backend feature API extensions.

### PR DESCRIPTION
**Pull request by acbeni**
A support for our backend tree.
Closes #1653

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/extensions/Extension.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @HasUpdateAction
+    List<Trigger> getTriggers();
+
+    @IgnoreInQueryModel
+    @HasUpdateAction
+    Destination getDestination();
+
+    @Override
+    @Nullable
+    @IgnoreInQueryModel
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/extensions/Extension.java]: Why do we ignore this property in our query model? In our other resources we don't ignore the key.

File: commercetools-models/src/main/java/io/sphere/sdk/extensions/TriggerType.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.sphere.sdk.extensions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public enum TriggerType {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/extensions/TriggerType.java]: Here we should use our standard mapping and implement the `SphereEnumeration` interface, this then allows us to remove the `JsonProperty` annotatio. But requires that implement the static method `ofSphereValue` (for an example see  `StateType`).

File: commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            Destination destination = HttpDestinationBuilder.of(AZURE_FUNCTION_URL, AzureFunctionsAuthenticationBuilder.of(randomKey()).build()).build();
+            Extension updatedExtension = client().executeBlocking(ExtensionUpdateCommand.of(extension, ChangeDestination.of(destination)));
+            assertThat(updatedExtension.getDestination()).isEqualTo(destination);
+            return updatedExtension;
+        });
+
+    }
+
+
+    public void withExtension(final BlockingSphereClient client, final ExtensionDraft extensionDraft, UnaryOperator<Extension> mapper) {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java]: `final` is missing for the last parameter.

File: commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+        Extension extension = client.executeBlocking(ExtensionCreateCommand.of(extensionDraft));
+        Extension result = mapper.apply(extension);
+        client.executeBlocking(ExtensionDeleteCommand.of(result));
+    }
+
+    public void withExtension(final BlockingSphereClient client, UnaryOperator<Extension> mapper) {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java]: `final` is missing for the last parameter.

File: commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SubscriptionFixtures.java [hunk 5]
```diff
@@ -69,6 +71,25 @@ public static void assumeHasAzureSBEnv() {
...
+    }
+
+
+    /**
+     *
+     * @return The azure function used to validate the resource lifecycle (create, delete ....)
+     * @see io.sphere.sdk.extensions.Extension
+     */
+    public static String azureFunctionUrl(){
+        final String ironMqUriEnv = System.getenv(AZUREFUNCTION_URL);
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SubscriptionFixtures.java]: Do we then add the `AZUREFUNCTION_URL` environment variable to our inhouse TeamCity build server?

Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/subscriptions/SubscriptionFixtures.java]: @katmatt yes, this was added to teamcity.

File: commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java (hunk 4 above)
Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java]: 👍 good catch

File: commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java (hunk 3 above)
Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/extensions/commands/ExtensionUpdateCommandIntegrationTest.java]: 👍 good catch

File: commercetools-models/src/main/java/io/sphere/sdk/extensions/TriggerType.java (hunk 2 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/extensions/TriggerType.java]: @katmatt Here the problem is that sphere enum starts with an uppercase, nonetheless i changed it to sphere enum by making the necessary overrides.

File: commercetools-models/src/main/java/io/sphere/sdk/extensions/Extension.java (hunk 1 above)
Comment (acbeni) [reply commercetools-models/src/main/java/io/sphere/sdk/extensions/Extension.java]: @katmatt  good catch 👍

**Comment by acbeni** [comment]
@katmatt all the changes have been taken care of 👍

**Comment by katmatt** [comment]
Looks fine 👍

**Pull request merged**